### PR TITLE
Add filesystem navigation

### PR DESCRIPTION
**Pull request by mrmacete**
Mount the io plugin as a file system:

```
[0x00000000]> m /mnt io
Mounted io on /mnt at 0x0
[0x00000000]> ms /mnt
```

On iOS, the filesystem is "virtualized" to quickly access relevant locations:

```
[/mnt]> ls
d AppHome
d AppBundle
d Device
[/mnt]> cd AppHome
[/mnt/AppHome]> ls
d Documents
f .com.apple.mobile_container_manager.metadata.plist
d Library
d SystemData
d tmp
[/mnt/AppHome]>
```

On other platforms you go directly to the root:

```
[0x00000000]> ms /mnt
[/mnt]> ls
d .HFS+ Private Directory Data
d home
d usr
d .Spotlight-V100
d net
f .DS_Store
d .PKInstallSandboxManager
d .PKInstallSandboxManager-SystemSoftware
d bin
f installer.failurerequests
d Network
d sbin
f .file
l etc
l var
d Library
d .Trashes
d System
f .OSInstallerMessages
d .fseventsd
d private
d .DocumentRevisions-V100
d .vol
d Users
d Applications
d opt
d dev
d Volumes
l tmp
d cores
[/mnt]>
```

Of course, depending on what process r2frida is attaching to a different subset of the file system is accessible according to its privileges.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2321,6 +2321,7 @@
       "resolved": "https://registry.npmjs.org/frida-compile/-/frida-compile-8.0.1.tgz",
       "integrity": "sha512-e9eAocCqRkLIIFyDHhrj3uR0gENNXuUuPG0NKHQNi1lN1SYpYPAVvIY6X06kwd34Fcr8cByZKUdfoUUvo3/EwA==",
       "requires": {
+        "babelify": "^10.0.0",
```
Comment (trufae) [inline package-lock.json]: that's a devDependency

File: package-lock.json [hunk 2]
```diff
@@ -2266,7 +2266,7 @@
     "find-root": {
       "version": "1.1.0",
       "resolved": "https://registry.npmjs.org/find-root/-/find-root-1.1.0.tgz",
-      "integrity": "sha512-NKfW6bec6GfKc0SGx1e07QZY9PE99u0Bft/0rzSD5k3sO/vwkVUpDUKVm5Gpp5Ue3YfShPFTX2070tDs5kB9Ng==",
+      "integrity": "sha1-q8/Iunb3CMQql7PWhbfpRQv7nOQ=",
```
Comment (trufae) [inline package-lock.json]: sha512 > sha1

File: src/agent/fs.js [hunk 3]
```diff
@@ -0,0 +1,563 @@
+'use strict';
+
+const {normalize} = require('path');
+const {platform, pointerSize} = Process;
+
+module.exports = {
+  ls,
+  cat,
+  open
```
Comment (trufae) [inline src/agent/fs.js]: maybe add `rm` too?

File: package-lock.json (hunk 2 above)
Comment (mrmacete) [reply package-lock.json]: oh, this is package-lock, how to control this?

**Comment by trufae** [comment]
from my experience its completely random, i also found some regressions switching from https to http randomly. which i fixed by hand. didnt checked where was the issue

**Pull request merged**